src/test/java/org/mule/extensions/jms/test/util/Eventually.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2023 Salesforce, Inc. All rights reserved.
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.extensions.jms.test.util;
+
+import static java.lang.Thread.sleep;
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+import org.mule.tck.probe.Timeout;
+
+import org.hamcrest.BaseMatcher;
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+
+/**
+ * Should the matcher be satisfied with delay?
+ */
+public class Eventually<T> extends BaseMatcher<T> {
+
+  private final Matcher<T> matcher;
+  private static final int TIMEOUT_SECS = 10;
+  private static final long RETRY_INTERVAL = 100;
+
+  /**
+   * Creates a matcher that retry the matcher until it is satisfied or until the timeout.
+   *
+   * @param matcher the matcher that will be executed until the timeout
+   */
+  public static <T> Eventually<T> eventually(Matcher<T> matcher) {
+    return new Eventually<>(matcher);
+  }
+
+  private Eventually(Matcher<T> matcher) {
+    this.matcher = matcher;
+  }
+
+  @Override
+  public boolean matches(Object item) {
+    Timeout timeout = new Timeout(SECONDS.toMillis(TIMEOUT_SECS));
+    while (!matcher.matches(item)) {
+      if (timeout.hasTimedOut()) {
+        return false;
+      }
+      try {
+        sleep(RETRY_INTERVAL);
```

Comment:
It is just a comment, but [Sonarqube does not recommend using the sleep instruction in the tests](https://rules.sonarsource.com/java/RSPEC-2925/) to avoid that the tests to be performed may work differently depending on the environment.

Reply:
In this case we are implementing a prober, similar to what `Awaitility` does, but much simpler. It is not just a single sleep for waiting. We set a total timeout we are willing the test to last at most and then check at regular (much smaller) intervals (the sleep).

I thought it looked nicer to read like this:
`assertThat(extensionClassLoader, is(eventually(collectedByGc())));`
rather than:
`await().until(() -> extensionClassLoader, is(collectedByGc()));`

But it is not that bad either. If you want I can change it. This module already depends on Awaitility, and I could remove the `Eventually` class completely.

Reply:
No, for me, this is clear and simple, don't change it only for the Sonarqube suggestion.